pygeobase/io_base.py

```diff
@@ -393,15 +568,19 @@ def write_gp(self, gpi, data, **kwargs):
         data : numpy.ndarray
             Data records.
         """
-        self._open(gpi)
-
         if self.mode in ['r']:
             raise IOError("File is not open in write/append mode")
 
+        self._open(gpi)
+
         lon, lat = self.grid.gpi2lonlat(gpi)
 
         self.fid.write_ts(gpi, data, lon=lon, lat=lat, **kwargs)
 
+        if self._write_bulk:
```

Comment:
Why would we flush and close if `write_bulk` is enabled? Should it not be the other way round?